### PR TITLE
refactor: unify tests in test apps

### PR DESCRIPTION
**Pull request by alduzy**
## Description

This PR intents to unify tests used by test apps

## Changes

- moved tests to common directory
- removed duplicates

## Checklist

- [ ] Included code example that can be used to test this change
- [ ] Updated TS types
- [ ] Updated documentation: 
  - [ ] https://github.com/software-mansion/react-native-screens/blob/main/guides/GUIDE_FOR_LIBRARY_AUTHORS.md
  - [ ] https://github.com/software-mansion/react-native-screens/blob/main/native-stack/README.md
  - [ ] https://github.com/software-mansion/react-native-screens/blob/main/src/types.tsx
  - [ ] https://github.com/software-mansion/react-native-screens/blob/main/src/native-stack/types.tsx
- [ ] Ensured that CI passes

### CODE REVIEW

**Review by maciekstosio — APPROVED** [review]

File: FabricTestExample/App.js
Comment (kkafar) [inline FabricTestExample/App.js:1]: We want to have App.js files in each application. This is usual RN application entry point and I would prefer expected application structure to remain.

So we would import only appropriate test files from shared module.

Comment (kkafar) [reply FabricTestExample/App.js:1]: Ideally we want it structured as it is [done in `reanimated`](https://github.com/software-mansion/react-native-reanimated/blob/main/FabricExample/App.tsx)

**Review by kkafar — COMMENTED** [review]
Good job overall. Left important remark

**Review by kkafar — COMMENTED** [review]
Oh, and there is one more thing. There could be some test cases that were present only in `FabricTestExample` and not in `TestsExample` or the other way around. Please make sure, that we do not remove them entirely.

File: FabricTestExample/App.js
Comment (alduzy) [reply FabricTestExample/App.js:1]: https://github.com/software-mansion/react-native-screens/pull/2135/commits/b4748213f217d903418746b141e524322650c1c3

File: FabricTestExample/App.js [hunk 1]
```diff
@@ -1,106 +1,3 @@
...
-import Test2048 from './src/Test2048';
-import Test2069 from './src/Test2069';
-import Test2118 from './src/Test2118';
-
-enableFreeze(true);
-
-export default function App() {
-  return <Test42 />;
-}
+export default App;
```
Comment (kkafar) [inline FabricTestExample/App.js:3]:
```suggestion
export default App;

```

File: TestsExample/App.js [hunk 2]
```diff
@@ -1,107 +1,3 @@
...
-import Test2048 from './src/Test2048';
-import Test2069 from './src/Test2069';
-import Test2118 from './src/Test2118';
-
-enableFreeze(true);
-
-export default function App() {
-  return <Test42 />;
-}
+export default App;
```
Comment (kkafar) [inline TestsExample/App.js:3]:
```suggestion
export default App;

```

**Review by kkafar — COMMENTED** [review]
Yeah, it looks nice now :tada:

At last we need better name for the `tests` directory, as this is usually used in context of unit/integration tests and gives wrong clues about its contents. What do you think of `test-screens` or something similar?

File: TestsExample/App.js (hunk 2 above)
Comment (alduzy) [reply TestsExample/App.js]: https://github.com/software-mansion/react-native-screens/pull/2135/commits/f588f7b4d9caf39d66c21169505bce3a2f579ed8

File: FabricTestExample/App.js (hunk 1 above)
Comment (alduzy) [reply FabricTestExample/App.js]: https://github.com/software-mansion/react-native-screens/pull/2135/commits/f588f7b4d9caf39d66c21169505bce3a2f579ed8

**Review by kkafar — APPROVED** [review]
I think we're good to go 🟢 
If you tested & everything is working I think we can proceed.

**Pull request merged**